packages/openapi3/test/shared-routes.test.ts

```diff
@@ -92,4 +92,30 @@ describe("openapi3: shared routes", () => {
       enum: ["resourceGroup", "subscription"],
     });
   });
+
+  it("emits diagnostic if shared parameters are of different types", async () => {
+    const diagnostics = await diagnoseOpenApiFor(
+      `
+      @service({title: "My Service"})
+      namespace Foo {
+        model Resource {
+          id: string;
+        }
+
+        @route("/sharedroutes/resources", { shared: true })
+        op listByResourceGroup(...Resource, @query filter: "resourceGroup"): Resource[];
+
+        @route("/sharedroutes/resources", { shared: true })
+        op listBySubscription(...Resource, @header filter: "subscription"): Resource[];
+      }
+      `
+    );
+    expectDiagnostics(diagnostics, [
+      {
+        code: "@typespec/openapi3/inconsistent-parameter-type",
+        message:
+          "Parameter 'filter' shared by multiple routes cannot be different types. Found: query, header.",
```

Comment:
I think we should refer to this as "location" and not "type". Because the type tripped me up on the parameter type not the "in".

Reply:
I mean, but if it is actually valid shouldn't we just allow it?

Reply:
Given that this is the OAI3 emitter, I guess we should. For some reason, I misread which openapi version we were fixing when initially reviewing the code (thus my edited example of what is allowed). For OAI2, I wouldn't bother fixing it since I don't think I've seen an actual Azure service relying on this. For OAI3, it would be better to fix. But not high priority IMHO, so if it is difficult, I would open a separate issue to address in the future.

Reply:
This logic isn't in the autorest emitter. It doesn't do any merging of anything--just dumps stuff into different x-ms-paths